### PR TITLE
Add ATenOp avg_pool2d and adaptive_avg_pool2d

### PR DESCRIPTION
**Pull request by iK1D**
Add ATenOp avg_pool2d and adaptive_avg_pool2d to support torchvision models including Alexnet, VGG11_bn, Squeezenet, Densenet and Inception v3.

### CODE REVIEW

**Review by SherlockNoMad — APPROVED** [review]

**Comment by SherlockNoMad** [comment]
Great! How far are we from running these few models ?

**Comment by iK1D** [comment]
> Great! How far are we from running these few models ?

For training, with Min's BatchNormInternal and fix of the parse schema in eval mode, we will be able to run all those models. For eval, we still need exporter to fix the Conv-BN fusion issue.

But the training perf is not as good as PT. Still need more investigation.

**Comment by SherlockNoMad** [comment]
Yes, we will still have a lot of perf tuning to do, especially for Conv and ConvGrad. 
Let's work on coverage first, and worry about perf latter. :)
